packages/ui/src/components/QueryBuilder/QueryPills/BooleanQueryPill.tsx

```diff
@@ -0,0 +1,72 @@
+import React from 'react';
+
+import StackLayout from '../../../layout/StackLayout';
+import { ISyntheticSqon, IValueFilter, TSqonGroupOp } from '../../../data/sqon/types';
+import { isBooleanOperator, isReference } from '../../../data/sqon/utils';
+import ReferenceQueryPill from './ReferenceQueryPill';
+import FieldQueryPill from './FieldQueryPill';
+import Combiner from '../Combiner';
+import { IDictionary, TCallbackRemoveAction, TCallbackRemoveReferenceAction } from '../types';
+
+interface IBooleanQueryPillProps {
+    parentQueryId: string;
+    query: ISyntheticSqon | Record<string, never>;
+    isActive: boolean;
+    showLabels?: boolean;
+    dictionary: IDictionary;
+    onRemoveFacet: TCallbackRemoveAction;
+    onRemoveReference: TCallbackRemoveReferenceAction;
+    onCombineChange?: (id: string, combinator: TSqonGroupOp) => void;
+    getColorForReference?: (refIndex: number) => string;
+}
+
+const BooleanQueryPill: React.FC<IBooleanQueryPillProps> = (props) => {
```

Comment:
please remove React.FC. I know it was used before, but we had a discussion to not use it anymore

see https://github.com/facebook/create-react-app/pull/8177